### PR TITLE
Check return value for GetModuleFileName

### PR DESCRIPTION
**Pull request by xiaoyinl**
This is a recreated pull request of PR #2757, as I messed up the commits there.

This patch checks the return values of calls to GetModuleFileName in ConfigParser.cpp. I am not so sure about how to test this. I saw most test cases in the `test` folder are js files, but this change is not related with JavaScript functionality.

### CODE REVIEW

**Comment by dilijev** [comment]
@suwc I think we can take this change without additional test cases, as this sort of thing is fairly difficult to test. If we decide we need to add tests later, we can do that.

File: lib/Common/Core/FaultInjection.cpp [hunk 1]
```diff
@@ -1228,16 +1228,20 @@ namespace Js
         GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS, reinterpret_cast<LPCTSTR>(ip), &mod);
         offset = ip - (uintptr_t)mod;
         auto& faultModule = modulePath;
-        GetModuleFileName(mod, faultModule, MAX_PATH);
-        fwprintf(stderr, _u("***FI: Exception: %08x, module: %s, offset: 0x%p\n"),
-            ep->ExceptionRecord->ExceptionCode, faultModule, (void*)offset);
+        if (GetModuleFileName(mod, faultModule, MAX_PATH))
+        {
+            fwprintf(stderr, _u("***FI: Exception: %08x, module: %s, offset: 0x%p\n"),
+                ep->ExceptionRecord->ExceptionCode, faultModule, (void*)offset);
```
Comment (dilijev) [inline lib/Common/Core/FaultInjection.cpp]: Is it okay to continue execution if `GetModuleFileName` call fails above?

File: lib/Common/Core/FaultInjection.cpp [hunk 2]
```diff
@@ -1228,16 +1228,20 @@ namespace Js
...
         uintptr_t savedOffset = 0;
         auto& mainModule = modulePath;
-        GetModuleFileName(NULL, mainModule, MAX_PATH);
-        // multiple session of Fault Injection run shares the single crash offset recording file
-        _snwprintf_s(filename, _TRUNCATE, _u("%s.FICrashes.txt"), mainModule);
+        if (GetModuleFileName(NULL, mainModule, MAX_PATH))
+        {
+            // multiple session of Fault Injection run shares the single crash offset recording file
+            _snwprintf_s(filename, _TRUNCATE, _u("%s.FICrashes.txt"), mainModule);
+        }
```
Comment (dilijev) [inline lib/Common/Core/FaultInjection.cpp]: Is it okay to continue execution if `GetModuleFileName` call fails above?

File: lib/Common/Core/ConfigParser.cpp [hunk 3]
```diff
@@ -513,7 +519,10 @@ HRESULT ConfigParser::SetOutputFile(const WCHAR* outputFile, const WCHAR* openMo
 
     char16 fileName[_MAX_PATH];
     char16 moduleName[_MAX_PATH];
-    GetModuleFileName(0, moduleName, _MAX_PATH);
+    if (!GetModuleFileName(0, moduleName, _MAX_PATH))
+    {
+        AssertMsg(FALSE, "Get filename of the current executable failed");
```
Comment (dilijev) [inline lib/Common/Core/ConfigParser.cpp]: Is it okay to continue execution if `GetModuleFileName` fails in release builds (where Asserts don't hit)?

**Comment by suwc** [comment]
@dilijev I agree with you.

---
In reply to: [305615082](https://github.com/Microsoft/ChakraCore/pull/3082#issuecomment-305615082) [](ancestors = 305615082)

**Comment by obastemur** [comment]
My 2 cents; `GetModuleFileName` (cross platform) has some issues. `GetBinaryLocation` should replace that. See #2297 .

File: lib/Common/Core/FaultInjection.cpp (hunk 1 above)
Comment (xiaoyinl) [inline lib/Common/Core/FaultInjection.cpp]: The buffer faultModule that GetModuleFileName fills is not used after the fwprintf, so I guess it's fine to only skip fwprintf and continue, when it fails.

**Comment by dilijev** [comment]
@obastemur May be true, but I think that's out of scope of this change (unless you disagree?). Let's continue to track that discussion separately.

**Comment by obastemur** [comment]
>  May be true, but I think that's out of scope of this change (unless you disagree?)

Well, what's the scope of this change?

This update is redundant and makes the future update unnecessarily complicated. There is an open Github issue with a direction and reason.

**Comment by dilijev** [comment]
The scope of this change is what the title says.

We can hold off if you think the alternative is going to make progress soon, however #2297 hasn't seen any discussion in nearly a month, and in general, we'd like to merge PRs from external folks without unnecessary delays.

This PR strictly improves the state things by checking the return values. I don't especially see why this PR makes a future change more complicated.

**Comment by obastemur** [comment]
@xiaoyinl Thanks for the effort. I talked @dilijev offline, we reached to a common understanding.

How `GetModuleFileName` used in ChakraCore, we don't need the return value. The replacement function is designed this in mind and already takes care of other cases.

If you want to help us on making `GetModuleFileName` usage right, I would like to have a chat with you on gitter about #2297 . Feel free to ping me on gitter so we can start talking on this (you can find me at @obastemur)

I added some details on fixing #2297, hope they are sufficient. See https://github.com/Microsoft/ChakraCore/issues/2297#issue-198068489 for implementation details.

**Comment by xiaoyinl** [comment]
Thank you @dilijev, @suwc and @obastemur!

@obastemur: Yes, I'd like to work on #2297. I will leave you a message on gitter. Thank you!

**Comment by obastemur** [comment]
Closing this one. #3112 is done, we will be moving to GetBinaryLocation instead. @xiaoyinl Thanks again! 👍

**Pull request closed without merging**